core/arch/arm/kernel/boot.c

```diff
@@ -118,30 +118,34 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
-#if defined(_CFG_CORE_STACK_PROTECTOR)
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
 /* Generate random stack canary value on boot up */
-__weak uintptr_t plat_get_random_stack_canary(void)
+__weak void plat_get_random_stack_canaries(void *buf, size_t blen, size_t clen)
```

Comment:
The relationship between blen and clen isn't fully clear. How about using the `qsort()`/`bsearch()` approach instead with `size_t nmemb, size_t size` instead? No big deal though, but the parameters must be fully described.